File: 414-third-maximum-number/414-third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
      
         Arrays.sort(nums);
        int n = nums.length;
        if(n < 3){
            return nums[n-1];
        }
        int k = 1;
        while(k < 3 && n>1){
            n--;
            if(nums[n] != nums[n-1]){
                k++;
            }   
        }
        if(k<3){
            return nums[nums.length-1];
        }
        return nums[n-1];
    }
}

File: 414-third-maximum-number/414-third-maximum-number.cpp
class Solution {
public:
  int thirdMax(vector<int>& nums) {
      if(nums.size()==1) return nums[0];
        if(nums.size()==2) return nums[0]>nums[1] ? nums[0] : nums[1];
        long first=LONG_MIN;
        long second=LONG_MIN;
        long third = LONG_MIN;
        for(int i=0;i<nums.size();i++){
             int current = nums[i];
            if(current>first){
                third=second;
                second=first;
                first=current;
            }
            else if(current>second && current!=first  ){
                third=second;
                second=current;
                
            }
            else if(current>third && current!=second && current!=first){
                third=current;
               
            }
            
        }
         return third==LONG_MIN ? first : third;
        
    }
};